### PR TITLE
get, count, functions and test DB:FS - gitignore

### PR DESCRIPTION
**Pull request by JoelRamirez2016**
plase review this pull request to continue

### CODE REVIEW

File: models/engine/db_storage.py [hunk 1]
```diff
@@ -74,3 +74,15 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
+        """retrieve one object"""
+        if isinstance(id, str):
+            for obj in self.all(cls).values():
+                if id in obj.id:
```
Comment (jsebdev) [inline models/engine/db_storage.py:82]: In this line 'in' should be '=='

File: models/engine/file_storage.py [hunk 2]
```diff
@@ -68,3 +68,15 @@ def delete(self, obj=None):
     def close(self):
         """call reload() method for deserializing the JSON file to objects"""
         self.reload()
+
+    def get(self, cls, id):
+        """retrieve one object"""
+        if isinstance(id, str):
+            for obj in self.all(cls).values():
+                if id in obj.id:
```
Comment (jsebdev) [inline models/engine/file_storage.py:76]: Also in this line 'in' should be '=='

**Review by jsebdev — APPROVED** [review]
Good addition. 
I did some comments on the code you can check to change 'in's for '=='s

**Comment by AlfredPianist** [comment]
Nice testing Joel. I have a question. Isn't more isolated to test the actual results of the `get` and `count` bypassing directly the methods `get` and `count`? For example, in the file_storage tests, you test the get and count methods by retrieving the items directly from the file and asserting them with the actual methods, and for the db_storage ones you test them by using the MySQLdb module and do the same? What do you think?

[This is the link to my GitHub repo.](https://github.com/AlfredPianist/AirBnB_clone_v3/tree/master/tests/test_models/test_engine) Check it out if you want to know what I mean.

**Pull request merged**